tests/network_bridge_test.py

```diff
@@ -0,0 +1,185 @@
+        kubeutils.assert_pod_state(
+            pod['metadata']['name'],
+            namespace=namespace,
+            state='Pending',
+            timeout=300
+        )
+
+    # Configure networks across cluster nodes
+    with _configure_host_networking('node01'), \
+            _configure_host_networking('node02'):
+
+        # Check that resources are finally reported as allocatable
+        for node, resource in itertools.product(
+            ['node01', 'node02'],
+            ['bridge.network.kubevirt.io/mybr1',
+             'bridge.network.kubevirt.io/mybr2']
+        ):
+            kubeutils.assert_resource_ready(
+                resource,
+                node=node,
+                timeout=60
+            )
+
+        # Check that pods were scheduled and successfully started
+        for pod in consumer_pods['items']:
+            kubeutils.assert_pod_state(
+                pod['metadata']['name'],
+                namespace=namespace,
+                state='Running',
+                timeout=300
+            )
+            kubeutils.assert_pod_exec_ready(
+                pod['metadata']['name'],
+                namespace=namespace,
+                timeout=60
+            )
+
+        # Configure addresses on pod interfaces connected to bridge resource
+        _set_address_on_resource_interface(
+            pod='bridge-consumer-1',
+            namespace=namespace,
+            resource='bridge.network.kubevirt.io/mybr1',
+            address='192.168.1.21/24'
+        )
+        _set_address_on_resource_interface(
+            pod='bridge-consumer-4',
+            namespace=namespace,
+            resource='bridge.network.kubevirt.io/mybr1',
+            address='192.168.1.24/24'
+        )
+
+        # Check if there is connectivity between two pods running on different
+        # nodes
+        interface = _get_interface_of_resource(
+            pod='bridge-consumer-1',
+            namespace=namespace,
+            resource='bridge.network.kubevirt.io/mybr1'
+        )
+        netutils.assert_ping(
+            pod='bridge-consumer-1',
+            namespace=namespace,
+            destination='192.168.1.24',
+            interface=interface
+        )
+
+
+@contextmanager
+def _configure_host_networking(node):
+    # On KubeVirt CI nodes, forwarding is disabled by default, enable it
+    nodes.ssh(node, ['sudo', 'iptables', '--policy', 'FORWARD', 'ACCEPT'])
+
+    # Create two VLAN interfaces on top of eth0 and connect them to separate
+    # bridges. That will give us two isolated networks between nodes
+    for net_id in [1, 2]:
+        nodes.ssh(node, """
+sudo ip link add link eth0 name eth0.{net_id} type vlan id {net_id} &&
+sudo ip link add name mybr{net_id} type bridge &&
+sudo ip link set eth0.{net_id} master mybr{net_id} &&
+sudo ip link set eth0.{net_id} up &&
+sudo ip link set mybr{net_id} up
+        """.format(net_id=net_id).split())
+    try:
+        yield
+    finally:
```

Comment:
This is not safe, you may have leftovers if part of the command above fail before even entering the try block.